packages/nimble-components/src/banner/specs/README.md

```diff
@@ -0,0 +1,167 @@
+  - `action-href` - if set, the action element will be a link with the given `href`. (Otherwise, when `action-text` is set, a button element will be displayed.)
+  - `action-button-appearance` - either `outline` or `ghost` (default). When an action button is displayed, this controls the appearance variant.
+- *Methods*
+- *Events*
+  - `close` - fired when the banner is dismissed
+  - `action` - fired when the action button is activated (by mouse or keyboard). This is the intended way for clients to perform an action.
+- *CSS Classes and CSS Custom Properties that affect the component*
+
+### Anatomy
+
+```html
+<div class="icon">
+  ${when(x => x.type === 'error', html`<nimble-icon-exclamation-mark></nimble-icon-exclamation-mark>`)}
+  ${when(x => x.type === 'warning', html`<nimble-icon-triangle-filled></nimble-icon-triangle-filled>`)}
+  ${when(x => x.type === 'info', html`<nimble-icon-info></nimble-icon-info>`)}
+</div>
+<div class="text">
+  <span class="heading">
+    ${x => x.heading}
+  </span>
+  ${x => x.text}
+</div>
+<div class="action">
+  ${when(x => x.actionText && x.actionHref, html`
+    <nimble-anchor href="${x => x.actionHref}">${x => x.actionText}</nimble-anchor>`
+  )}
+  ${when(x => x.actionText && !x.actionHref, html`
+    <nimble-button appearance="${x => x.actionButtonAppearance}">${x => x.actionText}</nimble-button>`
+  )}
+</div>
+  <div class="close">
+  ${when(x => !x.preventDismiss), html`
+    <nimble-button appearance="ghost" content-hidden>
+      <nimble-icon-xmark slot="start"></nimble-icon-xmark>
+    </nimble-button>`
+  )}
+  </div>
+```
+
+### Angular integration
+
+An Angular wrapper will be created for `nimble-banner`. No `ControlValueAccessor` is needed.
+
+`routerLink` will not be supported for the link.
+
+### Blazor integration
+
+Blazor integration for the `nimble-banner` will be provided.
+
+### Visual Appearance
+
+See XD document link at the top of this document.
+
+The biggest issue is that we will have to completely re-style the icons, link, and buttons:
+- Primary icon (error, warning, or info) color is transparent white (in all themes)
+- Action button height is 24px (instead of 32px)
+- Action button hover border color is white (in all themes)
+- Action link color is white (in all themes)
+- Close button is 16px square
+- Close button icon is 8px square
+- Close button hover effect is background color change (transparent white) rather than border
+
+We are not able to directly style some of these elements, because they are in the shadow DOM of  `nimble-icon-*` and `nimble-button`. We will have to resort to overriding Nimble token values, like `--ni-nimble-icon-color` and `--ni-nimble-icon-size`.
+
+---
+
+## Implementation
+
+### States
+
+N/A
+
+### Accessibility
+
+*Consider the accessibility of the component, including:*
+
+- *Keyboard Navigation and Focus*
+    - the action button/link and the close button will be focusable tab stops
```

Comment:
How will we provide a title for the dismiss button? Since we don't localize text in Nimble, does this need to be another attribute on the API?

Reply:
Good point. It would be nice if there were some language-neutral way to communicate a close control, but I guess that doesn't exist. I suppose we have to make it an attribute...

Reply:
`title='🙅‍♀️'`?

😀